riscv-page-tables/src/sv48x4.rs

```diff
@@ -95,4 +95,29 @@ impl PagingMode for Sv48x4 {
         let num_l4_pages = 4;
         num_l1_pages + num_l2_pages + num_l3_pages + num_l4_pages
     }
+
+    fn max_huge_pte_pages(num_pages: u64, page_size: PageSize) -> u64 {
```

Comment:
I'm having a hard time making sense of this. Is `num_pages` in units of `page_size`, or is it always 4kB? I'm skeptical of the math either way -- take for example 4MB of 4kB pages (`num_pages = 1024`). This looks like it will return 4 PTE pages, which is most definitely incorrect.

This would be a lot easier to follow if we specified the amount of address space in number of bytes (or start + end) rather than number of pages (see comment above).